June 2021 news (#18275)
idaholab/moose: modules/doc/content/newsletter/2021_08.md

the same home address. Think of this feature as the ability to combine many websites into one. For
example, the main [MOOSE website](index.md exact=True) and the
[workshop/index.md alternative=missing_config.md] can now be built together even though they are
translated from MooseDown to HTML in a much different way and require separate styling and
JavaScript methods. Another example is two pages could use two different main menu bars or different
executables and instructions for retrieving [application syntax](MooseDocs/extensions/appsyntax.md).

Most notably, this new feature provides the ability to cross-link between the content sets using the
[python/MooseDocs/extensions/autolink.md]. The build routines assume that there's no difference
between pages of different configurations, including while making edits during a live reload, and
the user can assume that interactions between pages work exactly the same as they always have. The
[MooseDocs/config.md] page was revised to reflect these changes, namely, please see the
[MooseDocs/config.md#multiconfigs] section to learn how to take advantage of this new system.

## Improvements to Navier Stokes Flow capabilities

> - Thermal boundary conditions, including radiative and convective heat transfer boundary conditions
>   were reworked to support primitive, mixed and conservative variable sets for fluid flow.

Reviewer: ```suggestion
- Thermal boundary conditions, including radiative and convective heat transfer boundary conditions,
  were reworked to support primitive, mixed, and conservative variable sets for fluid flow.
```

- When using time integrators deriving from [ExplicitTimeIntegrator.md], if
  `solve_type` was set to anything other than `LINEAR`, a segmentation fault
  would occur. Now, an error message is given instead.
- Missing class documentation was added for actions and auxiliary kernels.
- When a block or boundary is mispelled, the code will now output the name of the
  block or boundary in addition to a newly generated id, which will simplify the
  search for the issue.

> - Dependendy resolution between user objects and auxiliary kernels will now be cognizant

Reviewer: ```suggestion
- Dependency resolution between user objects and auxiliary kernels will now be cognizant
```

  of their respective execution flags.

> - If the dependency resolution between UOs and auxiliary kernel is known to fail in a given case,

Reviewer: ```suggestion
- If the dependency resolution between user objects and auxiliary kernels is known to fail in a given case,
```
